usersession/agent/rest_api_test.go

```diff
@@ -903,3 +906,111 @@ func (s *restSuite) TestPostCloseRefreshNotification(c *C) {
 		"desktop-entry": dbus.MakeVariant("io.snapcraft.SessionAgent"),
 	})
 }
+
+func createDesktopFile(info *snap.AppInfo, icon string) {
+	data := []byte("[Desktop Entry]\nName=" + info.Name + "\n")
+	if icon != "" {
+		data = append(data, []byte("Icon="+icon+"\n")...)
+	}
+	os.MkdirAll(path.Dir(info.DesktopFile()), 0755)
+	os.WriteFile(info.DesktopFile(), data, 0644)
+
+}
+
+func createSnapInfo(snapName string) *snap.Info {
+	si := snap.Info{
+		SideInfo: snap.SideInfo{
+			RealName: snapName,
+		},
+		Apps: make(map[string]*snap.AppInfo, 5),
+	}
+	return &si
+}
+
+func addAppToSnap(snapinfo *snap.Info, app string, isService bool, icon string) {
+	newInfo := snap.AppInfo{
+		Snap: snapinfo,
+		Name: app,
+	}
+	if isService {
+		newInfo.Daemon = "daemon"
+	}
+	snapinfo.Apps[app] = &newInfo
+	createDesktopFile(&newInfo, icon)
+}
+
+func (s *restSuite) TestGuessAppIconNoIconPrefixEqualApp(c *C) {
+	si := createSnapInfo("app1")
+	addAppToSnap(si, "app1", false, "")
+	icon := agent.GuessAppIcon(si)
+	c.Check(icon, Equals, "")
+}
+
+func (s *restSuite) TestGuessAppIconNoIconPrefixDifferentApp(c *C) {
+	si := createSnapInfo("snap1")
+	addAppToSnap(si, "app1", false, "")
+	icon := agent.GuessAppIcon(si)
+	c.Check(icon, Equals, "")
+}
+
+func (s *restSuite) TestGuessAppIconPrefixDifferentApp(c *C) {
+	si := createSnapInfo("snap1")
+	addAppToSnap(si, "app1", false, "iconname")
+	icon := agent.GuessAppIcon(si)
+	c.Check(icon, Equals, "iconname")
+}
+
+func (s *restSuite) TestGuessAppIconPrefixEqualApp(c *C) {
+	si := createSnapInfo("app1")
+	addAppToSnap(si, "app1", false, "iconname")
```

Comment:
maybe add another app for good measure?
```suggestion
	addAppToSnap(si, "app1", false, "iconname1")
	addAppToSnap(si, "app2", false, "iconname1")
```

Reply:
Done.